manifests/params.pp

```diff
@@ -19,8 +19,17 @@
     $datadir    = '/etc/puppetlabs/puppet/hieradata'
     $owner      = 'pe-puppet'
     $group      = 'pe-puppet'
-    $provider   = 'pe_gem'
     $confdir    = '/etc/puppetlabs/puppet'
+    if $::puppetversion =~ /3.7/ {
+      $provider = 'pe_puppetserver_gem'
+      # 3.7 also needs /opt/puppet/bin/eyaml
+      exec { 'install pe_gem':
```

Comment:
Yeah, that's definitely a conundrum.  There's gotta be a good solution, but for the life of me I can't figure one out at the moment.